backend/pkg/songsLib/songs.go

```
@@ -110,9 +154,82 @@ func (s *SongGetter) GetSoundcloudDetails(url string) (SongDetails, error) {
+		fmt.Println("failed to parse youtube url: %w", err)
+		return
+	}
+
+	track_id := u.Path
+	//TODO: test how long lings last for, currently 24 hours for testing 
+	s.insertSongInCache(track_id, song_url, "soundcloud", time.Now().Add(time.Hour*24).Unix(), song_details)
+}
+
+
+func (s *SongGetter) insertSongInCache(song_id string, song_url string, source string, expires_at int64, song_details SongDetails) {
+	s.ss.InsertSong(&models.SongMapping{
```

golangci-lint (1.62.2)

Error return value of `s.ss.InsertSong` is not checked

(errcheck)